[Ripple] Additional documentation for Ripple and its classes (#6996)
material-components/material-components-ios: components/Ripple/README.md

```
<!--</div>-->

The `MDCRippleTouchControllerDelegate` gives you some control over aspects of the
ripple/touch relationship and its placement in the view hierarchy.

> In the below example we are using the delegate to declare we want to only process ripple 

Reviewer: ```suggestion
In the example below we are using the delegate to declare that we only want to process ripple 
```

> touches if the touch is in a certain location, and also insert the Ripple view at the very bottom of

Reviewer: ```suggestion
touches if the touch is in a certain location. We also insert the Ripple view at the bottom of
```

> the parent view's view hierarchy:

Reviewer: Consider explaining why we need to insert the ripple view at the bottom of the parent view's hierarchy here. Good opportunity to educate.

```swift
let myCustomView = MyCustomView(frame: CGRect.zero)
let rippleView = MDCRippleView()
rippleView.rippleColor = UIColor.red
myCustomView.addSubview(rippleView)
...
// When the touches begin, there is one animation
rippleView.beginRippleTouchDownAtPoint(at: touchPoint, animated: true, completion: nil)
...
// When the touches end, there is another animation
rippleView.beginRippleTouchUpAnimated(animated: true, completion: nil)
```

#### Objective-C
```objc
MyCustomView *myCustomView = [[MyCustomView alloc] initWithFrame:CGRectZero];
MDCRippleView *rippleView = [[MDCRippleView alloc] init];
rippleView.rippleColor = [UIColor redColor];
[myCustomView addSubview:rippleView];
...
// When the touches begin, there is one animation
[rippleView beginRippleTouchDownAtPoint:touchPoint animated:YES completion:nil];
...
// When the touches end, there is another animation
[rippleView beginRippleTouchUpAnimated:YES completion:nil];
```
<!--</div>-->

### MDCStatefulRippleView

You can also use MDCStatefulRippleView to display stateful ripples using your
own touch processing. 

> To fully benefit from MDCStatefulRipple's capabilities in moving through states visually,

Reviewer: ```suggestion
To fully benefit from MDCStatefulRipple's ability to move between states visually,
```

> the view that is adding the stateful ripple view will need to override 

Reviewer: ```suggestion
the view that is adding the stateful ripple view must override 
```

UIView's `touchesBegan`, `touchesMoved`, `touchesEnded` and `touchesCancelled`

> and call the stateful ripple views corresponding APIs before calling the super.

Reviewer: ```suggestion
and call the stateful ripple view's corresponding APIs before calling the super.
```

> and call the stateful ripple views corresponding APIs before calling the super.

Reviewer: ```suggestion
and call the stateful ripple views corresponding APIs before calling the `super` implementation.
```
